src/ssz/type/bit_list.zig

```diff
@@ -57,6 +58,27 @@ pub fn BitList(comptime limit: comptime_int) type {
             }
         }
 
+        pub fn getTrueBitIndexes(self: *const @This(), allocator: std.mem.Allocator, out: *[]usize) !void {
+            var buffer = try allocator.alloc(usize, self.bit_len);
+            defer allocator.free(buffer);
+            var true_bit_count: usize = 0;
+
+            for (self.data.items, 0..) |byte, byte_index| {
+                const bits = try computeByteToBitBooleanArray(byte);
+
+                for (0..8) |bit_index| {
+                    const overall_index = byte_index * 8 + bit_index;
+                    if (bits[bit_index]) {
+                        buffer[true_bit_count] = overall_index;
+                        true_bit_count += 1;
+                    }
+                }
+            }
+
+            out.* = try allocator.alloc(usize, true_bit_count);
+            @memcpy(out.*, buffer[0..true_bit_count]);
```

Comment:
this is a really weird pattern, I don't think this is right.

If we're going to allocate a slice of usize for the output, just return it.
eg
```zig
pub fn getTrueBitIndexes(self: *const @This(), allocator: std.mem.Allocator) ![]usize {
    ...
    const out = try allocator.alloc(usize, true_bit_count);
    @memcpy(out, buffer[0..true_bit_count]);
    return out;
}
```

If we're going to pass in a slice for the output, then don't allocate over it.
eg:
```zig
pub fn getTrueBitIndexes(self: *const @This(), out: *[]usize) !usize {
    if (out.len < self.bit_len) return error.TooShort;
    ...
    if (bits[bit_index]) {
        out[true_bit_count] = overall_index;
        true_bit_count += 1;
    }
    ...
    return true_bit_count;
}
```

What we're doing here is a little of both.